File: Frontend/src/app/DTOs/Learn/LearnQuestionDTO.ts
import { GeneralQuestionQuestionDTO } from "../GeneralQuestion/GeneralQuestionQuestionDTO";

export interface LearnQuestionDTO {
    question: GeneralQuestionQuestionDTO;
    answered: boolean;
    answeredCorrect: boolean | null;
    markedAsHard: boolean;
    solution: string | null;
}

File: Backend/DTOs/Learn/LearnQuestionDTO.cs
namespace Learnz.DTOs;
public class LearnQuestionDTO
{
    public GeneralQuestionQuestionDTO Question { get; set; }
    public bool Answered { get; set; }
    public bool? AnsweredCorrect { get; set; }
    public bool MarkedAsHard { get; set; }
    public string? Solution { get; set; }
}